### PR TITLE
Delete button deletes item from list and firebase after user confirma…

### LINKED ISSUE #12: 12. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future
## Summary
Users might make a mistake when entering an item, or may decide they won’t be buying a certain item again in the future. Allowing them to delete list items will help them keep their lists tidy.
 
## Acceptance criteria
- [x] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [x] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [x]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Notes:
* Activating the “Delete” button should show a confirmation dialog. We want to give users a chance to confirm “destructive” actions.
* It might be tempting to create your own modal dialog component to handle confirmation, but modals are fraught with accessibility concerns, so consider using a simple [JavaScript confirm dialog](https://developer.mozilla.org/en-US/docs/Web/API/Window/confirm).

### PR DESCRIPTION
**Pull request by granolagabrielle**
## Description

This code renders a delete button for each list item. Once the user clicks the button, the app prompts them to confirm that they want to delete the item. Once approved, the item is removed from the list and the database. 

## Related Issue

closes #12 

## Acceptance Criteria

- [X] The `ListItem` component renders a button that allows the user to delete an item from their list when clicked
- [X] Clicking the delete button prompts the user to confirm that they *really* want to delete the item
- [X]  The `deleteItem` function in `api/firebase.js` has been filled out, and deletes the item from the Firestore database

## Type of Changes

enhancement

### Before
Screenshot 2024-09-16 at 6 07 19 PM

### After
Screenshot 2024-09-16 at 6 06 21 PM
Screenshot 2024-09-16 at 6 06 37 PM

## Testing Steps / QA Criteria

1. Click the delete button next to an item
2. Select "Ok" from the alert to confirm delete for that item
3. Check firebase to ensure the item was deleted
4. Also select "cancel" to ensure the item stays in the list

### CODE REVIEW

File: src/views/List.jsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export function List({ data }) {
 					<p>Welcome to {listName}!</p>
 					<p>Ready to add your first item? Start adding below!</p>
 
-					<AddItems />
+					<AddItems items={items} />
```
Comment (warrenchan13) [inline src/views/List.jsx:28]: based on the failed test,  i think you need to import items to use it as a prop for the AddItems component

**Review by warrenchan13 — APPROVED** [review]
I left one comment. Other than that, it looks good!

File: src/views/List.jsx (hunk 1 above)
Comment (kweeuhree) [reply src/views/List.jsx]: hey! thank you for catching that! i forgot to push the last changes, i think that this was breaking our deployed app, so i added this change

**Review by deeheber — APPROVED** [review]
This looks good to me and passes all of the acceptance criteria. Tested locally and on the deployed versions.

Going to let this PR sit for a little to see if another mentor also wants to review to get you extra feedback. Otherwise, I'll merge this end of day tomorrow (Sept 19). Let me know if you don't want that to happen.

**Pull request merged**